src/dhtproto/client/request/internal/Update.d

```diff
@@ -0,0 +1,792 @@
+                goto case Error;
+        }
+
+        assert(false);
+    }
+
+    /***************************************************************************
+
+        Starts a new request-on-conn to write the record value to another node.
+        Asks the node the record was read from to remove it; handles the
+        response code.
+
+        Params:
+            conn = connection to communicate over
+            second_node_addr = address/port of the node that the updated record
+                will be sent to
+
+        Returns:
+            true if the update succeeded, false if there was an error
+
+    ***************************************************************************/
+
+    private bool updateOnDifferentNode ( RequestOnConn.EventDispatcher conn,
+        AddrPort second_node_addr )
+    {
+        this.context.shared_working.first_request_on_conn = conn;
+        this.context.shared_working.second_node_addr = second_node_addr;
+
+        // Start a new request-on-conn to write to the other node. This will
+        // cause the static `handler()` function to be called again in a new
+        // RequestOnConn instance that can be attached to a different
+        // connection. The shared working data contains everything the second
+        // request-on-conn needs.
+        new_request_on_conn();
+
+        // Wait until the second request-on-conn finishes. (This is necessary
+        // because this ROC owns the acquired resources tracker, which contains
+        // the buffer for the updated value. If this ROC were to simply exit at
+        // this stage, this buffer would be relinquished.)
+        auto event = conn.nextEvent(conn.NextEventFlags.init);
+        assert(event.active == event.Active.resumed);
+        assert(event.resumed.code == Update.SecondROCFinished);
+
+        // If the request on the second request-on-conn failed, the result code
+        // will already have been set in the shared working data.
+        if ( this.context.shared_working.second_node_failed )
+            return false;
+
+        // Send message to the node that was read from, informing it that the
+        // record was successfully updated on another node and may be deleted.
+        conn.send(
+            ( conn.Payload payload )
+            {
+                payload.addCopy(MessageType.RemoveRecord);
+            }
+        );
+        conn.flush();
+
+        // Handle response message.
+        auto result = conn.receiveValue!(MessageType)();
+        with ( MessageType ) switch ( result )
+        {
+            case Ok:
+                return true;
+
+            case Error:
+                // The node returned an error code. Notify the user.
+                Update.Notification n;
+                n.node_error = RequestNodeInfo(this.context.request_id,
+                    conn.remote_address);
+                Update.notify(this.context.user_params, n);
+                return false;
```

Comment:
It won't be a problem because when the redistribution process moves the record from the old -> new node, it will be discarded as the record already exists on the new node.

Reply:
That's good, thanks for the clarification!